app/src/main/java/com/internship/firstweekapp/ui/game/ZerosCrossesGameBoardView.kt

```diff
@@ -0,0 +1,258 @@
+package com.internship.firstweekapp.ui.game
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.RectF
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+import java.lang.Float.min
+
+
+enum class CellState {
+    EMPTY, CROSS, ZERO
+}
+
+class ZerosCrossesGameBoardView @JvmOverloads constructor(
+    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
+) : View(context, attrs, defStyleAttr) {
+
+    private var empties = 9
+
+    private var _stateArr = arrayOf(
+        arrayOf(CellState.EMPTY, CellState.EMPTY, CellState.EMPTY),
+        arrayOf(CellState.EMPTY, CellState.EMPTY, CellState.EMPTY),
+        arrayOf(CellState.EMPTY, CellState.EMPTY, CellState.EMPTY)
+    )
+
+    private var stateArr: Array<Array<CellState>>
+        get() = _stateArr
+        set(value) {
+            _stateArr = value
+            invalidate()
+        }
+
+    private lateinit var playerFigure: CellState
+
+    var playerFigureIsCross = true
+        set(value) {
+            field = value
+            playerFigure = if (value) {
+                CellState.CROSS
+            } else {
+                CellState.ZERO
+            }
+        }
+
+    var playerIsFirst = true
+
+
+    init {
+        isClickable = true
+
+    }
+
+    private val mainPaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        style = Paint.Style.FILL
+        color = Color.GREEN
+    }
+
+    private val cellPaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        style = Paint.Style.FILL
+        color = Color.WHITE
+    }
+
+    private val figurePaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        color = Color.BLACK
+        style = Paint.Style.STROKE
+        strokeWidth = 10F
+    }
+
+    private val bigRedLinePaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        color = Color.RED
+        style = Paint.Style.STROKE
+        strokeWidth = 20F
+    }
+
+    private var w = 0f
+    private var h = 0f
+
+    override fun onSizeChanged(ww: Int, hh: Int, oldw: Int, oldh: Int) {
+        w = min(ww.toFloat(), hh.toFloat())
+        h = min(ww.toFloat(), hh.toFloat())
+    }
+
+    override fun onDraw(canvas: Canvas) {
+        super.onDraw(canvas)
+        canvas.drawRect(0f, 0f, w, h, mainPaint)
+
+        for (i in 0..2)
```

Comment:
0..2 should be not hardcoded
digging info about board dimension from Game class